File: Problems_and_Solutions/day-4/Java/Solution3.java
class Solution3{ 
    public static int findMaxConsecutiveOnes(int[] nums) {
        int count=0,ones=0;
        for(int i=0; i<nums.length; i++){
            if(nums[i] == 1){
                ones++;
            }
            else{
                ones=0;
            }
            count = Math.max(ones,count);
        }
        return count;
    }

    public static void main(String args[]){
        int[] arr = {1, 1, 0, 1, 1, 1};
        System.out.println(findMaxConsecutiveOnes(arr));
    }
}

File: Problems_and_Solutions/day-4/cpp/solution 3.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int max=0;
        int curCount=0;
        for(int i=0;i<nums.size();i++)
        {
            if(nums[i])
                curCount++;
            else
            {
                if(curCount>max)
                    max=curCount;
                curCount=0;
            }
            
        }  
        if(curCount>max)
            max=curCount;
        return max; 
    }
};